resource_customizations/kafka.banzaicloud.io/KafkaCluster/health.lua

```diff
@@ -3,18 +3,19 @@ if obj.status ~= nil then
   if obj.status.brokersState ~= nil then
     local counter = 0
     local brokerReady = 0
-    for i, broker in pairs(obj.status.brokersState) do
-        if (brokerReady <= tonumber(i)) then
-            brokerReady = tonumber(i)+1
-        else
-            brokerReady = brokerReady
-        end
-        if broker.configurationState == "ConfigInSync" and broker.gracefulActionState.cruiseControlState == "GracefulUpscaleSucceeded" then
-          counter = counter + 1
-        end
-        if broker.configurationState == "ConfigInSync" and broker.gracefulActionState.cruiseControlState == "GracefulDownscaleSucceeded" then
-          counter = counter + 1
-        end
+    for i, broker in ipairs(obj.status.brokersState) do
+      local brokerIndex = tonumber(i) - 1
+      if (brokerReady <= brokerIndex) then
+        brokerReady = brokerIndex+1
+      else
+        brokerReady = brokerReady
```

Comment:
I know this is old code...

Is this `else` assignment needed?

Related:
- Won't `brokerReady` always be equal to number of brokers e.g. `#obj.status.brokersState`?
- It looks like `counter` is intended to store number of ready brokers (based on checks) and `brokerReady` is intended to store number of total brokers, so the name of these variables look reversed. Is this intended?

Reply:
> else necessary 

Good catch! Didn't think too hard about it. 

> number of brokers 

Exactly, but the original loop logic appears to be trying to find the **max** broker id (which is not guaranteed to start at zero). This is causing the original issue to check that `counter == 104`, for example, so it's being marked as degraded for 3 otherwise healthy brokers with ids 101, 102, and 103.

I don't know Lua, actually, but upon research, yes `#obj.status.brokersState` seems more correct. I'll push that change today. Also, I see the degraded.yaml testdata is failing, and I wasn't able to clearly identify why.